### PR TITLE
Feat/read ini

### PR DESCRIPTION
**Pull request by ChristianLieven**
Enables the pre-commit hook to read the `memote.ini`.
Further allows the user to specify separate paths to writing either YAML, JSON or MAT files.

Fixes https://app.zenhub.com/workspace/o/opencobra/memote/issues/435

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -60,3 +60,6 @@ target/
 
 #Ipython Notebook
 .ipynb_checkpoints
+
+#PyCharm
+.idea
```
Comment (Midnighter) [inline .gitignore]: Please enter a newline here after `.idea`.

**Comment by Midnighter** [comment]
The format of the `memote.ini` is determined by `memote/suite/cli/config.py` so I'm not a fan of simply adding additional values to it here. Was there a request for other output formats? I don't really want to start supporting Matlab files and JSON is not a great format for diff'ing either.

**Comment by ChristianLieven** [comment]
Oh I didn't know about this:
> The format of the memote.ini is determined by memote/suite/cli/config.py

Will fix that accordingly.

Neither the MAT file nor the JSON file are for diffing, they are rather optional concessions to our respective userbases (COBRA-Toolbox vs. Cobrapy/ CobraJIL), who might feel more at home importing a model straight away in these formats respectively. The primary reason why I added this is to allow natively and periodically writing other files to cater for instance to the requirements of the yeast-gem maintainers.

I'm OK with removing them though if you disagree.

**Comment by Midnighter** [comment]
> I'm OK with removing them though if you disagree.

I'm afraid I do. I think that if you work in other tools then the workflow is anyway: Make changes to model -> create outputs from your favorite tool (MAT, JSON, SBML) -> commit -> YAML for diff'ing -> memote run. So I don't actually see it serving non-cobrapy users to create those outputs and we cannot (or I don't want to have to) guarantee that loading a model from SBML in cobrapy, creating an output in MAT and loading that MAT with the COBRA toolbox to again generate SBML is a fully loss-less cycle.

**Comment by Midnighter** [comment]
Fix #9

**Comment by ChristianLieven** [comment]
>So I don't actually see it serving non-cobrapy users to create those outputs and we cannot (or I don't want to have to) guarantee that loading a model from SBML in cobrapy, creating an output in MAT and loading that MAT with the COBRA toolbox to again generate SBML is a fully loss-less cycle.

I personally prefer to develop a model with the JSON format, particularly because then I can extend the model with non-canonical information I add to the model object internally. I remember Ben supporting his users by providing a MAT file for straight-forward use of the model without having the hassle of [conversion](https://github.com/opencobra/memote/issues/258#issuecomment-381560328). 

For publications, releases, model interchange, and testing in Memote, however, the SMBL is the one source of truth. I don't quite see though how we're harming that by being forthcoming with what Cobrapy offers if our message on that stays clear.

Ensuring a loss-less cycle with more than just SBML is an issue I can get behind, although cobrapy does offer all these formats, and I think you'd only reasonably be expected to maintain that half of the cycle.

**Pull request merged**